avo-hq/avo
# feature: expose stimulus hooks

# Description


This PR adds the ability for the user to hook into the resource pages and change the behavior through stimulus controllers.

Docs: https://docs.avohq.io/2.0/stimulus-integration.html


# Checklist:



## Manual review steps


1. Go to a course record
2. Toggle the `Has skills` checkbox. Observe how the skills field gets hidden and displayed
3. Change the Country select. Observe how the city select gets new values from the server

Manual reviewer: please leave a comment with output from the test if that's the case.

## Screenshots

![country-city-select](https://user-images.githubusercontent.com/1334409/172179186-04bce78c-1ec2-4b82-b6f5-741047f2623a.gif)
![debug-on-input](https://user-images.githubusercontent.com/1334409/172179216-1aec474b-0cf1-4391-95ce-681c8c8fc893.gif)
![disable-method](https://user-images.githubusercontent.com/1334409/172179218-9274374f-9ed3-4f99-83a0-74bbfc5ea91a.gif)
![toggle-method](https://user-images.githubusercontent.com/1334409/172179222-3415264c-ae20-4256-bca9-34fc7c2f3852.gif)
